dev-tools/omdb/src/bin/omdb/nexus.rs

```diff
@@ -253,6 +273,12 @@ impl NexusArgs {
                 omdb.check_allow_destructive()?;
                 cmd_nexus_sled_add(&client, args).await
             }
+            NexusCommands::Sleds(SledsArgs {
+                command: SledsCommands::Expunge(args),
+            }) => {
+                omdb.check_allow_destructive()?;
```

Comment:
Same comment as https://github.com/oxidecomputer/omicron/pull/5231#discussion_r1520283160 -- it would be nice to have a token here indicating that `check_allow_destructive` has been called.

Reply:
Implemented in 27812fde5. I'll let this land first, then use it in #5231 too.